src/fairchem/core/units/mlip_unit/predict.py

```diff
@@ -268,6 +277,13 @@ def predict(
                 pred_output[task_name] = task.normalizer.denorm(
                     output[task_name][task.property]
                 )
+                if self.assert_on_nans:
+                    assert (
+                        not torch.isnan(pred_output[task_name]).any()
```

Comment:
Are we distinguishing between nan and inf? otherwise we could use isfinite?
Distinguishing might be useful though..